### PR TITLE
Add health/readiness and health/liveness endpoint

### PR DESCRIPTION
**Pull request by walterddr**
Adding health/readiness and health/liveness endpoints
- readiness returns 200 when all data from a pinot server is loaded
- liveness returns 200 as soon as the REST application is started.

See: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/

this fixes: #9023

### CODE REVIEW

File: pinot-server/src/main/java/org/apache/pinot/server/api/resources/HealthCheckResource.java [hunk 1]
```diff
@@ -55,7 +55,36 @@ public class HealthCheckResource {
...
+    return getReadinessStatus(_instanceId);
+  }
+
+  @GET
+  @Path("/health/liveness")
+  @Produces(MediaType.TEXT_PLAIN)
+  @ApiOperation(value = "Checking server liveness status.")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Server is healthy"),
+      @ApiResponse(code = 503, message = "Server is not healthy")
```
Comment (Jackie-Jiang) [inline pinot-server/src/main/java/org/apache/pinot/server/api/resources/HealthCheckResource.java:67]: (minor)
```suggestion
      @ApiResponse(code = 200, message = "Server is live"),
      @ApiResponse(code = 503, message = "Server is not live")
```

File: pinot-server/src/main/java/org/apache/pinot/server/api/resources/HealthCheckResource.java [hunk 2]
```diff
@@ -55,7 +55,36 @@ public class HealthCheckResource {
...
+  }
+
+  @GET
+  @Path("/health/readiness")
+  @Produces(MediaType.TEXT_PLAIN)
+  @ApiOperation(value = "Checking server instance health")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Server is healthy"),
+      @ApiResponse(code = 503, message = "Server is not healthy")
+  })
```
Comment (Jackie-Jiang) [inline pinot-server/src/main/java/org/apache/pinot/server/api/resources/HealthCheckResource.java:81]: (minor)
```suggestion
  @ApiOperation(value = "Checking server readiness status")
  @ApiResponses(value = {
      @ApiResponse(code = 200, message = "Server is ready to serve queries"),
      @ApiResponse(code = 503, message = "Server is not ready to serve queries")
  })
```

File: pinot-server/src/main/java/org/apache/pinot/server/api/resources/HealthCheckResource.java [hunk 3]
```diff
@@ -55,7 +55,36 @@ public class HealthCheckResource {
       @ApiResponse(code = 503, message = "Server is not healthy")
   })
   public String checkHealth() {
```
Comment (Jackie-Jiang) [inline pinot-server/src/main/java/org/apache/pinot/server/api/resources/HealthCheckResource.java:57]: Consider adding a deprecate annotation with some comment

**Review by Jackie-Jiang — APPROVED** [review]

File: pinot-server/src/main/java/org/apache/pinot/server/api/resources/HealthCheckResource.java (hunk 3 above)
Comment (walterddr) [reply pinot-server/src/main/java/org/apache/pinot/server/api/resources/HealthCheckResource.java:57]: per discussion in https://github.com/apache/pinot/issues/9023 I think we should skill keep this. 
I will merge for now and if needed we can come up with a plan for deprecate.

**Pull request merged**